src/collections/Table/TableRow.js

```diff
@@ -76,6 +93,20 @@ TableRow.propTypes = {
   /** A row may call attention to an error or a negative value. */
   error: PropTypes.bool,
 
+  /** An element type to render as (string or function). */
+  itemAs: customPropTypes.as,
```

Comment:
OK, let's save it for a later PR.

Here's an idea for the record.  I have this same issue with `FeedExtra`.  It has the need for two different shorthands, one that uses `images` and one that uses `text`.  I'm thinking of just adding:

```jsx
FeedExtra.createImages()
FeedExtra.createText()
````

What if we added this?

```jsx
TableRow.create()
TableRow.createHeader()
````

The latter would set `as` on each element in the array.  I think this touches on how collection shorthands are different than an item shorthand.  A string or number is not valid in a collection, `TableRow.create('not valid') //=> <Table items='not valid' />`.  So it seems we don't need a factory that only provides a `mapValueToProps` but instead/also a way to update each item. Perhaps with a default props object/function somehow.  Again, this comment is for later reference.